openwisp_monitoring/monitoring/base/models.py

```diff
@@ -154,6 +181,9 @@ def write(self, value, time=None, database=None, check=True, extra_values=None):
         """ write timeseries data """
         values = {self.field_name: value}
         if extra_values and isinstance(extra_values, dict):
+            for key in extra_values.keys():
+                if key not in self.related_fields:
```

Comment:
I think this line will fail when self.related_fields is None because NoneType is not iterable

Reply:
I think failure is what's expected as `related_fields` shouldn't be **None** for a metric configuration receiving `extra_values`.

Reply:
Yes, but at the moment None is the returned value when the KeyError exception is risen

Reply:
Should be taken care of, now :smiley: